src/osd/PG.h

```diff
@@ -1120,7 +1120,7 @@ class PG : public DoutPrefixProvider, public PeeringState::PeeringListener {
     utime_t sleep_start;
 
     // flags to indicate explicitly requested scrubs (by admin)
-    bool must_scrub, must_deep_scrub, must_repair;
+    bool must_scrub, must_deep_scrub, must_repair, need_auto;
```

Comment:
i feel that `must_scrub` and `need_auto` are mutual-exclusive. can we consolidate them into a single variable? like
```c++
enum class Initiated {
  by_user,
  by_routine,
  on_error,
  none,
};
Initiated why = Initiated::none;
```

actually, the other flags can be populated from this single variable.

Reply:
@tchaikov need_auto may be superseded by a user request.  So must_scrub could become set in a rare case with a need_auto condition.  The latest code will ignore and then clear need_auto in that rare race case.